lib/chat_models/chat_open_ai.ex

```diff
@@ -148,11 +152,14 @@ defmodule LangChain.ChatModels.ChatOpenAI do
     Enum.map(functions, &ForOpenAIApi.for_api/1)
   end
 
-  defp set_response_format(%ChatOpenAI{json_response: true}),
-    do: %{"type" => "json_object"}
+  defp conditionally_add_response_format(output, %ChatOpenAI{model: "gpt-4-vision-preview"}), do: output
+
+  defp conditionally_add_response_format(output, %ChatOpenAI{json_response: true}),
+    do: Map.put(output, :response_format, %{"type" => "json_object"})
+
+  defp conditionally_add_response_format(output, %ChatOpenAI{json_response: false}),
+    do: Map.put(output, :response_format, %{"type" => "text"})
```

Comment:
Hopefully there's a way to detect images without pinning it to a specific beta model.

Reply:
Yeah, this is all pretty grose cause it's a beta. TBH, the fact that the beta doesn't support _any_ response_format is :spew: